auditlog/event_builder.go

```diff
@@ -0,0 +1,203 @@
+package auditlog
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/ory/gojsonschema"
+
+	"github.com/ory/oathkeeper/pipeline/authn"
+	"github.com/ory/oathkeeper/proxy"
+)
+
+// EventBuilder is a type to build the Event structure.
+type EventBuilder struct {
+	URLPattern string `json:"url_pattern"`
+	Method     string `json:"http_method"`
+	Filter     Filter `json:"filter"`
+	// Logger string `json:"logger"`
+	DescriptionTemplate string `json:"description_template"`
+
+	r *regexp.Regexp
+}
+
+type Filter struct {
+	RequestHeaderWhiteList  []string `json:"request_header"`
+	RequestBodyWhiteList    []string `json:"request_body"`
+	ResponseHeaderWhiteList []string `json:"response_header"`
+	ResponseBodyWhiteList   []string `json:"response_body"`
+}
+
+// UnmarshalJSON implements the Unmarshaler interface for the EventBuilder struct.
+func (b *EventBuilder) UnmarshalJSON(raw []byte) error {
+	var err error
+
+	// An additional struct like in rule.go.
+	var bb struct {
```

Comment:
could you please add some tests to show the usage of such construction? or provide more comments